libraries/chain/transaction_context.cpp

```diff
@@ -313,6 +321,13 @@ namespace bacc = boost::accumulators;
       is_initialized = true;
    }
 
+   void transaction_context::add_ram_provider(const action& action) {
```

Comment:
It's unexpected to get action as parameter to this function. I think better pass `provideram`.